src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,134 @@
-import React from 'react';
+import React, { useState, useMemo, useEffect } from 'react';
+import { addComment, deleteComment, getPostComments } from '../../api/comments';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type PostProps = {
+  postId: number,
+  content: string,
+};
+
+export const PostDetails: React.FC <PostProps> = ({
+  postId,
+  content,
+}) => {
+  // alert('Post details rendering');
+
+  const [
+    areCommentsVisible,
+    setAreCommentsVisible,
+  ] = useState(true);
+
+  const [
+    comments,
+    setComments,
+  ] = useState<Comment[]>([]);
+
+  const [
+    commentsChanged,
+    setCommentsChanged,
+  ] = useState(false);
+
+  const toggleCommentsVisibility = () => {
+    setAreCommentsVisible(prevState => !prevState);
+  };
+
+  const addNewComment = (
+    event: React.FormEvent<HTMLFormElement>,
+    currentPostId: number,
+    commentUserName: string,
+    commentEmail: string,
+    commentText: string,
+  ) => {
+    event.preventDefault();
+
+    addComment(currentPostId, commentUserName, commentEmail, commentText);
+    setCommentsChanged(prevState => !prevState);
+  };
+
+  const removeComment = (commentId: number) => {
+    deleteComment(commentId);
+    setCommentsChanged(prevState => !prevState);
+  };
+
+  useEffect(() => {
+    getPostComments(postId).then(commentsArr => setComments(commentsArr));
+  }, []);
+
+  useMemo(async () => {
+    const commentsArr = await getPostComments(postId);
+
+    setComments(commentsArr);
+  }, [commentsChanged, postId]);
```

Comment:
it's not how the useMemo used
useMemo is just a hook, it's not a lifecycle method

https://reactjs.org/docs/hooks-reference.html#usememo

Reply:
Ok